iceoryx_utils/include/iceoryx_utils/internal/units/duration.inl

```diff
@@ -21,47 +21,102 @@ namespace iox
 namespace units
 {
 template <typename T>
-constexpr Duration Duration::nanoseconds(const T ns)
+constexpr Duration Duration::nanoseconds(const T value)
 {
-    return operator"" _ns(ns);
+    static_assert(std::numeric_limits<T>::is_integer, "only integer are supported");
+
+    if (value < 0)
+    {
+        return Duration{0U, 0U};
```

Comment:
I think a warning that the duration is set to 0 would be good. Same for the other c'tors.